internal/grpc_dialer.go

```diff
@@ -109,7 +111,18 @@ func dial(params dialParameters) (*grpc.ClientConn, error) {
 }
 
 func requiredInterceptors(metricScope tally.Scope, headersProvider HeadersProvider) []grpc.UnaryClientInterceptor {
-	interceptors := []grpc.UnaryClientInterceptor{metrics.NewScopeInterceptor(metricScope), errorInterceptor}
+	interceptors := []grpc.UnaryClientInterceptor{
+		errorInterceptor,
+		// Report aggregated metrics for the call, this is done outside of the retry loop.
+		metrics.NewRPCMetricsInterceptor(metricScope, "_total"),
```

Comment:
Better create constant for "_total".
I would be careful using "_total" here since it implies aggregation over arbitrary subset of tags and it can become obscure if reused in a different location. For example, we already reuse "_total" to aggregate values across multiple namespaces on server.

In this case, my initial hunch was that we aggregate some metrics across all the different calls, while regular metrics will be per-something (like per API name).

It may be better to use empty suffix here, but add "_per_retry" on L124.

Reply:
Hm, swapping the two is an interesting idea. Let's do that. Then all users that are relying on existing metrics would get correct ones automatically.